### PR TITLE
Address CSV exporting OutOfMemory issue

### PR DESCRIPTION
**Pull request by mbriskar**
Instead of firstly gathering String[] and then writing, I now open the files and then sequentially write lines to them. I am not sure though if the problem is not also in Query.from(InlineHintModel.class) which gather Iterable of all the hints. No memory exception with test files that I tried (about 5).

### CODE REVIEW

File: reporting/tests/src/test/resources/test-exports/app2.export [hunk 1]
```diff
@@ -1,4 +1,4 @@
 "Rule Id";"Problem type";"Title";"Description";"Links";"Application";"File Name";"File Path";"Line";"Story points"
 "rule2";"hint";"hint2;""""""""-te""xt";"";"";"app2";"f2";"/f2";"0";"3"
-"classification2";"classification";"classification2-text";"";"";"app2";"f2";"/f2";"N/A";"300"
 "classification1";"classification";"classification1-text";"description-classification";"[link,description]";"app2";"f2";"/f2";"N/A";"30"
+"classification2";"classification";"classification2-text";"";"";"app2";"f2";"/f2";"N/A";"300"
```
Comment (mbriskar) [inline reporting/tests/src/test/resources/test-exports/app2.export]: Ordering of the classification in the .export file has changed, nothing else.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/332/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/332/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/332/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/961/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.reporting:windup-reporting-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/961/org.jboss.windup.reporting$windup-reporting-tests/testReport'>org.jboss.windup.reporting:windup-reporting-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/961/org.jboss.windup.reporting$windup-reporting-tests/testReport/org.jboss.windup.reporting/CSVExportingTest/testCSVExportGeneration/'><strong>org.jboss.windup.reporting.CSVExportingTest.testCSVExportGeneration</strong></a></li></ul>
Test FAILed.

File: reporting/impl/src/main/java/org/jboss/windup/reporting/export/ExportCSVFileRuleProvider.java [hunk 2]
```diff
@@ -60,7 +62,8 @@ public Configuration getConfiguration(GraphContext context)
             .addRule()
             .when(Query.fromType(WindupConfigurationModel.class).withProperty(WindupConfigurationModel.CSV_MODE,true)
                         .and(Query.fromType(ClassificationModel.class).as("classifications")
-                                    .and(Query.fromType(InlineHintModel.class).as("hints"))))
+                                    .and(Query.fromType(InlineHintModel.class).as("hints")
```
Comment (jsight) [inline reporting/impl/src/main/java/org/jboss/windup/reporting/export/ExportCSVFileRuleProvider.java]: I would like to see this moved into the operation itself. Rewrite does not do short-circuit evaluation, so these queries run even when export is not enabled. In the case of a large application, this is adding quite a bit of time to the windup run.

Comment (mbriskar) [inline reporting/impl/src/main/java/org/jboss/windup/reporting/export/ExportCSVFileRuleProvider.java]: ok. good idea

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/333/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/333/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/333/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/962/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/335/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/964/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.ui:windup-ui-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/964/org.jboss.windup.ui$windup-ui-tests/testReport'>org.jboss.windup.ui:windup-ui-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/964/org.jboss.windup.ui$windup-ui-tests/testReport/org.jboss.windup.addon.ui/WindupCommandTest/org_jboss_windup_addon_ui_WindupCommandTest/'><strong>org.jboss.windup.addon.ui.WindupCommandTest.org.jboss.windup.addon.ui.WindupCommandTest</strong></a></li></ul>
Test FAILed.

**Comment by mbriskar** [comment]
is that fialing test connecting to this ?

**Comment by mbriskar** [comment]
retest please

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/339/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/339/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/339/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

File: reporting/impl/src/main/java/org/jboss/windup/reporting/export/ExportCSVFileRuleProvider.java [hunk 3]
```diff
@@ -72,97 +75,110 @@ public Configuration getConfiguration(GraphContext context)
...
-                if (hint.getFile() != null)
+                ProjectModel root = project.getRootProjectModel();
+                if (!rootProjectWriters.containsKey(root.getName()))
                 {
-                    if (hint.getFile().getProjectModel() != null)
+                    try
                     {
-                        projectNameString = hint.getFile().getProjectModel().getName();
+                        CSVWriter writer = new CSVWriter(
+                                    new FileWriter(config.getOutputPath().getFilePath() + File.separator + root.getName() + ".csv"), ';');
```
Comment (lincolnthree) [inline reporting/impl/src/main/java/org/jboss/windup/reporting/export/ExportCSVFileRuleProvider.java]: Good. I think .csv makes more sense.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/968/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.reporting:windup-reporting-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/968/org.jboss.windup.reporting$windup-reporting-tests/testReport'>org.jboss.windup.reporting:windup-reporting-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/968/org.jboss.windup.reporting$windup-reporting-tests/testReport/org.jboss.windup.reporting/CSVExportingTest/testCSVExportGeneration/'><strong>org.jboss.windup.reporting.CSVExportingTest.testCSVExportGeneration</strong></a></li></ul>
Test FAILed.

**Pull request merged**